azure-mgmt/tests/test_common.py

```diff
@@ -42,21 +44,21 @@ def __init__(self, credentials, subscription_id, base_url):
                 self.base_url = base_url
 
         temp_auth_file = tempfile.NamedTemporaryFile(delete=False)
-        temp_auth_file.write(configuration)
+        temp_auth_file.write(json.dumps(configuration).encode())
```

Comment:
Laziness, since I know my test content is ASCII, so I won't get issue anywhere. But yes, I should specify an actual encoding :p